modules/core/src/test/java/org/apache/synapse/endpoints/RecipientListEndpointTest.java

```diff
@@ -0,0 +1,154 @@
+public class RecipientListEndpointTest {
+
+    /**
+     * Test on init of RecipientListEndpoint
+     */
+    @Test
+    public void testRecipientListEndpointInit() {
+        try {
+            Axis2SynapseEnvironment synapseEnvironment = getMockedSynapseEnvironment();
+            RecipientListEndpoint recipientListEndpoint = new RecipientListEndpoint();
+
+            recipientListEndpoint.init(synapseEnvironment);
+        } catch (Exception ex) {
+            Assert.fail("recipientListEndpoint init must be successful without exception " + ex);
+        }
+    }
+
+    /**
+     * Test on Sending messages to different members using RecipientListEndpoint
+     */
+    @Test
+    public void testSendMessageToMemembers() {
+        try {
+            //perform init
+            Axis2SynapseEnvironment synapseEnvironment = getMockedSynapseEnvironment();
+            RecipientListEndpoint recipientListEndpoint = new RecipientListEndpoint();
+            recipientListEndpoint.init(synapseEnvironment);
+            Mockito.when(synapseEnvironment.createMessageContext()).thenReturn(createMessageContext());
+
+            //set members
+            Member member1 = new Member("localhost", 9000);
+            Member member2 = new Member("localhost", 9001);
+            ArrayList <Member> members = new ArrayList<>(2);
+            members.add(member1);
+            members.add(member2);
+            recipientListEndpoint.setMembers(members);
+
+            //test send message
+
+            Axis2MessageContext messageContext = getMessageContext();
+            //message will be sent to EP using this env (which is mocked and do nothing)
+            messageContext.setEnvironment(synapseEnvironment);
+            messageContext.setTo(new EndpointReference("http://localhost:9000/services/SimpleStockQuoteService"));
+            recipientListEndpoint.sendMessage(messageContext);
+        } catch (Exception ex) {
+            Assert.fail("recipientListEndpoint should be able to send message to members without exception " + ex);
+        }
+    }
+
+    /**
+     * Create a sample synapse message context with a simple payload
+     *
+     * @return Axis2MessageContext with payload and parameters
+     * @throws Exception on creating the context
+     */
+    private Axis2MessageContext getMessageContext() throws Exception {
+        String samplePayload = "<test>value</test>";
+        Map<String, Entry> properties = new HashMap<>();
+        Axis2MessageContext messageContext = TestUtils.getAxis2MessageContext(samplePayload, properties);
+        messageContext.getAxis2MessageContext().setTransportIn(new TransportInDescription("http"));
+        return messageContext;
+    }
+
+    /**
+     * Create a mock SynapseEnvironment object
+     *
+     * @return Axis2SynapseEnvironment instance
+     * @throws AxisFault on creating/mocking object
+     */
+    private Axis2SynapseEnvironment getMockedSynapseEnvironment() throws AxisFault {
```

Comment:
This method looks like a reusable Util method. Will it help if we move it to a common place?

Reply:
Depending on the test the deed of the method calls to the mock object needs to be modified. So there is not much use moving it to a common place.